tsdb/head.go

```diff
@@ -307,6 +309,35 @@ func newHeadMetrics(h *Head, r prometheus.Registerer) *headMetrics {
 	return m
 }
 
+// HeadStats are the statistics for the head component of the DB.
+type HeadStats struct {
+	WALReplayStatus *WALReplayStatus
+}
+
+// NewHeadStats returns a new HeadStats object.
+func NewHeadStats() *HeadStats {
+	return &HeadStats{
+		WALReplayStatus: &WALReplayStatus{},
+	}
+}
+
+// WALReplayStatus contains status information about the WAL replay.
+type WALReplayStatus struct {
+	sync.RWMutex
+	Min     int
+	Max     int
+	Current int
+}
+
+// GetWALReplayStatus returns the WAL replay status information.
+func (s *WALReplayStatus) GetWALReplayStatus() WALReplayStatus {
+	s.RLock()
+	x := *s
+	s.RUnlock()
+
+	return x
```

Comment:
This is still failing although there is nothing wrong given the WALReplayStatus does not  have pointers inside itself (that would not work):

```
GO111MODULE=on /go/bin/golangci-lint run --timeout 2m ./...
tsdb/head.go:335:7: copylocks: assignment copies lock value to x: github.com/prometheus/prometheus/tsdb.WALReplayStatus (govet)
	x := *s
	     ^
tsdb/head.go:338:9: copylocks: return copies lock value: github.com/prometheus/prometheus/tsdb.WALReplayStatus (govet)
	return x
```

I would suggest explicit copy then. Create new `WALReplayStatus` and copy fields manually. WDYT?

Reply:
Sure!